app/services/carrierwave_filename.rb

```diff
@@ -2,7 +2,7 @@ module CarrierwaveFilename
   def for(attachment, underscore: false)
     file = attachment.file && attachment.file.path.split('/').last
 
-    underscore ? file.tr('-', '_') : file
+    underscore ? file.gsub(/[^a-zA-Z0-9](?=[[:alnum:]]*\.)/, '_') : file
```

Comment:
Yeah totes agree. Sorry missed this comment, was considering another way to achieve the same result that conveys our intent a little better.